packages/platform-sdk/test/integration-tests/CleanEverythingIntegrationTest.ts

```diff
@@ -0,0 +1,242 @@
+import { Product } from '../../src'
+import { apiRoot } from './test-utils'
+
+// Generic function to handle deletion of resources
+const deleteResource = async (resourceType, resourceId, version) => {
+  try {
+    await apiRoot[resourceType]()
+      .withId({ ID: resourceId })
+      .delete({ queryArgs: { version } })
+      .execute()
+    console.log(`Deleted ${resourceType.slice(0, -1)} with ID: ${resourceId}`)
+  } catch (error) {
+    console.error(
+      `Error deleting ${resourceType.slice(0, -1)} with ID: ${resourceId}:`,
+      error
+    )
+  }
+}
+
+// Function to unpublish and delete a product
+const unpublishAndDeleteProduct = async (product: Product) => {
+  if (product.masterData.published) {
+    console.log(`Unpublishing product: ${product.id}`)
+    await apiRoot
+      .products()
+      .withId({ ID: product.id })
+      .post({
+        body: {
+          version: product.version,
+          actions: [{ action: 'unpublish' }],
+        },
+      })
+      .execute()
+    console.log(`Product ${product.id} unpublished.`)
+  }
+
+  // Increment version after unpublish
+  await deleteResource(
+    'products',
+    product.id,
+    product.version + (product.masterData.published ? 1 : 0)
+  )
+}
+
+// Function to delete all products for a specific product type
+const deleteProductsForProductType = async (productTypeId: string) => {
+  const products = await apiRoot
+    .products()
+    .get({ queryArgs: { where: `productType(id="${productTypeId}")` } })
+    .execute()
+
+  console.log(
+    `Found ${products.body.results.length} products for product type ${productTypeId}...`
+  )
+  return Promise.all(
```

Comment:
It's not recommended to use `Promise.all` with requests to commercetools. Rather use [Promise.map](http://bluebirdjs.com/docs/api/promise.map.html) and limit the concurrency to some lower number like 20. The reason is CTP cannot handle well the deletion and deletion usually triggers some backend work as well. I know this because I did `Promise.all` as well and caused the whole platform outage in the past 😅